test/reverse.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+var throws = require('assert').throws;
+
+var eq = require('./utils').eq;
+var errorEq = require('./utils').errorEq;
+var S = require('..');
+
+
+describe('slice', function() {
+
+  it('is a unary function', function() {
+    eq(typeof S.reverse, 'function');
+    eq(S.reverse.length, 1);
+  });
+
+  it('reverses arrays', function() {
+    eq(S.reverse([1, 2, 3]), [3, 2, 1]);
+    eq(S.reverse(['1', '2', '3']), ['3', '2', '1']);
+    eq(S.reverse([]), []);
+  });
+
+  it('reverses strings', function() {
+    eq(S.reverse('123'), '321');
+    eq(S.reverse(''), '');
+  });
+
+  it('type checks its arguments', function() {
```

Comment:
Our convention is to test invalid applications before valid applications. Would you mind moving this above the `'reverses arrays'` block?